### PR TITLE
Allow nested types with kotlinx package

### PR DESCRIPTION
**Pull request by asodja**
Also add more information why a type is not supported with Nested.

Fixes https://github.com/gradle/gradle/issues/26018

### CODE REVIEW

File: subprojects/model-core/src/main/java/org/gradle/internal/properties/annotations/NestedValidationUtil.java [hunk 1]
```diff
@@ -51,33 +52,29 @@ public static void validateBeanType(
...
+        ));
     }
 
-    private static boolean isGString(Class<?> type) {
-        return type.getName().startsWith("groovy.lang.GString");
-    }
-
-    private static boolean isSupportedType(Class<?> type) {
-        return !isJavaSE(type) && !isKotlinStdlib(type) && !isGString(type);
+    private static Optional<String> isSupportedType(Class<?> type) {
```
Comment (ov7a) [inline subprojects/model-core/src/main/java/org/gradle/internal/properties/annotations/NestedValidationUtil.java:68]: 💅 Consider renaming this to something like `getUnsupportedReason`. The current name does not align well with the return type.

**Review by ov7a — APPROVED** [review]

File: subprojects/model-core/src/main/java/org/gradle/internal/properties/annotations/NestedValidationUtil.java (hunk 1 above)
Comment (asodja) [reply subprojects/model-core/src/main/java/org/gradle/internal/properties/annotations/NestedValidationUtil.java]: Makes sense

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test and merge

**Pull request merged**